L1Trigger/L1THGCal/src/be_algorithms/HGCalMulticlusteringImpl.cc

```diff
@@ -39,39 +39,26 @@ bool HGCalMulticlusteringImpl::isPertinent( const l1t::HGCalCluster & clu,
 }
 
 
-bool HGCalMulticlusteringImpl::isNeighbor( const l1t::HGCalCluster & clu1, 
-                                           const l1t::HGCalCluster & clu2) const
-{
-  HGCalDetId cluDetId( clu2.detId() );
-  HGCalDetId firstClusterDetId( clu1.detId() );
-  
-  if( cluDetId.zside() != firstClusterDetId.zside() ){
-    return false;
+void HGCalMulticlusteringImpl::findNeighbor( const std::vector<std::pair<int,double>>&  rankedList,
+                                             unsigned int searchInd,
+                                             const edm::PtrVector<l1t::HGCalCluster> & clustersPtrs, 
+                                             std::vector<int>& neighbors
+                                            ){
+
+  for(unsigned int ind = searchInd+1; ind < rankedList.size() && fabs(rankedList.at(ind).second - rankedList.at(searchInd).second) < distDbscan_ ; ind++){
+    if(((*(clustersPtrs[rankedList.at(ind).first])).centreProj() - (*(clustersPtrs[rankedList.at(searchInd).first])).centreProj()).mag() < distDbscan_){
```

Comment:
Something should be done in case `clustersPtrs[...]` is out of bound.

Reply:
The added code doesn't seem to protect from `rankedList.at(ind).first` or `rankedList.at(searchInd).first` to be out of `clustersPtrs` bounds.